include/psuedocode.md

```diff
@@ -0,0 +1,54 @@
+The follow outlines some psuedocode for how to implement the logic for a prototype QEO solution.
+
+```
+struct offload_info {
+    uint operation;
+    uint direction;
+    uint next_packet_number;
+    addr ip_address;
+    byte cid[];
+    key_info key;
+}
+
+offload_info[] rx_offload_table;
+offload_info[] tx_offload_table;
+
+int rx_cid_length = -1;
+int tx_cid_length = -1;
+
+void update_offload(offload_info offloads[]) {
+    foreach (auto offload in offloads) {
+        if (offload.operation == "add") {
+            if (offload.direction == "tx") {
+                if (tx_cid_length == -1) tx_cid_length = offload.cid.length;
+                tx_offload_table.add_or_update(offload);
+            } else {
+                if (rx_cid_length == -1) rx_cid_length = offload.cid.length;
+                rx_offload_table.add_or_update(offload);
+            }
+        } else {
+            if (offload.direction == "tx") {
+                tx_offload_table.remove(offload);
+            } else {
+                rx_offload_table.remove(offload);
+            }
+        }
+    }
+}
+
+void on_packet_tx(packet packet) {
+    if (packet.udp_payload[0] == "long header") return;
+    byte[] cid = packet.udp_payload[1 .. tx_cid_length+1];
+    uint packet_number = packet.udp_payload[tx_cid_length+2 .. tx_cid_length+6]; // 4 bytes
+    offload_info offload = tx_offload_table.find(packet.ip_address, cid);
```

Comment:
I thought we agreed the TX path needs to use trusted OOB data, not header data, to find the connection offload info, including secret key?

Reply:
Not for the prototype, which is specified as the goal for this up top. There is no OOB data for prototype.

Reply:
Oh, or is this specifically for the early prototype design?

Reply:
Gotcha